modules/abstract-eth/src/abstractEthLikeNewCoins.ts

```diff
@@ -33,20 +78,241 @@ interface HopPrebuild {
   gasPriceMax: number;
 }
 
+/**
+ * The extra parameters to send to platform build route for hop transactions
+ */
+interface HopParams {
+  hopParams: {
+    gasPriceMax: number;
+    userReqSig: string;
+    paymentId: string;
+  };
+  gasLimit: number;
+}
+
+export interface EIP1559 {
+  maxPriorityFeePerGas: number;
+  maxFeePerGas: number;
+}
+
+export interface ReplayProtectionOptions {
+  chain: string | number;
+  hardfork: string;
+}
+
 export interface TransactionPrebuild extends BaseTransactionPrebuild {
   hopTransaction?: HopPrebuild;
   buildParams: {
     recipients: Recipient[];
   };
   recipients: TransactionRecipient[];
-  nextContractSequenceId: string;
+  nextContractSequenceId: number;
   gasPrice: number;
   gasLimit: number;
   isBatch: boolean;
   coin: string;
   token?: string;
 }
 
+export interface SignFinalOptions {
+  txPrebuild: {
+    eip1559?: EIP1559;
+    replayProtectionOptions?: ReplayProtectionOptions;
+    gasPrice?: string;
+    gasLimit?: string;
+    recipients?: Recipient[];
+    halfSigned?: {
+      expireTime: number;
+      contractSequenceId: number;
+      backupKeyNonce?: number;
+      signature: string;
+      txHex?: string;
+    };
+    nextContractSequenceId?: number;
+    hopTransaction?: string;
+    backupKeyNonce?: number;
+    isBatch?: boolean;
+    txHex?: string;
+    expireTime?: number;
+  };
+  signingKeyNonce?: number;
+  walletContractAddress?: string;
+  prv: string;
+  recipients?: Recipient[];
```

Comment:
Here I had to make all the params optional because I was getting an error that the params passed to signTransaction are not compatible as we also have signTransaction in AbstractEthLikeCoin class and the interface used there is not having all these fields.

Reply:
Moving from required to optional isnt a breaking change right?

Reply:
Correct, will remove this from breaking change